### ISSUE TITLE
Add example to connect with CausalPy geo-lift?

### ISSUE THREAD

**Issue by thipokKub** [pymc-labs/pymc-marketing#255]
Lift test is important for MMM model calibration, but it is I'm not sure on how to update prior to the experiment lift test. Recently pymc just released [CausalPy](https://causalpy.readthedocs.io/en/latest/), and I think it is a good idea to have an example on how to predict, evaluate, and iterate on the results

**Comment by twiecki** [comment]
Yes, we do plan to add this. Is this something you are interested in contributing?

**Comment by thipokKub** [comment]
I'm not sure, I don't know how exactly on a couple things
- How to add MMM model to [CausalPy](https://github.com/pymc-labs/CausalPy), but likely to do with [this script template](https://github.com/pymc-labs/CausalPy/blob/main/causalpy/pymc_models.py)
- Not sure on how to iterate model after incremental test
    - In [Lightweight_mmm](https://github.com/google/lightweight_mmm), they didn't specify how to update their model after incremental testing. This can be seen missing from their flowchart (below)
    ![flowchart](https://raw.githubusercontent.com/google/lightweight_mmm/main/images/flowchart.png)
    - In [Robyn](https://facebookexperimental.github.io/Robyn/docs/analysts-guide-to-MMM#modeling-techniques-)
        - in *Model Calibration* section, they have optimize using `NRMSE` , and `decomp.RSSD`
        -  `robyn_refresh()` which will create new model consistent with previous data, and new data
        - Robyn has the option to extract trend + seasonality from the whole data, (using prophet) while only modeled using small window of media data (e.g. `2` years of data collection, but using only `3` months of media spending, see *Model Window* section)

In Robyn, I think it is because they use optimization based approach, they can customize, and update with the new data. But for Bayesian in PyMC, other than restart the whole model, I don't see how to update model to be consistent with the incremental testing results

**Comment by drbenvincent** [comment]
Hi @thipokKub. Would you be able to just expand a bit on your thoughts here? The issue title seems to focus on CausalPy and geolift (there is an example on that in the causalpy docs https://causalpy.readthedocs.io/en/latest/notebooks/geolift1.html), but the main content of the issue seems to focus more on how to incorporate lift test observations into the MMM.

**Comment by thipokKub** [comment]
Sorry, I've got a bit off-track. What I was trying to do is finding a repeatable testing, and calibration pipeline for MMM model, which is not exactly what I asked. But from what I've found from [this pdf](https://analytic-edge.com/wp-content/uploads/2022/10/The-Value-of-Calibrating-MMM-with-Lift-Experiments.pdf). The model need to be able to perform lift test, then recalibrated from the observed ROAS

I'm thinking that CausalPy can be used in for either synthetic control or difference-in-difference analysis specifically on ROAS/mROAS

**Comment by drbenvincent** [comment]
I believe the lift test/calibration part of this is best suited for an issue in [pymc-marketing](https://github.com/pymc-labs/pymc-marketing).

We have an aspiration that CausalPy and PyMC-Marketing will synergise, and this sounds like it could be a particular example of that. Right now, we are not quite there, but I'm hoping to spend some time soon to ensure that we've got the right abstractions in CausalPy to be used by PyMC-Marketing.

Tagging @cluhmann, @lucianopaz, @ricardoV94 for their thoughts.

**Comment by drbenvincent** [comment]
**UPDATE: 2nd Feb 2026**

## National level lift tests

We now have [pymc-marketing docs](https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_lift_test.html#about-lift-tests) that shows how to do lift test calibration for national level models. This is supported by [docs on the causalpy side](https://causalpy.readthedocs.io/en/latest/notebooks/its_lift_test.html) of things, focussing specifically on the national level model case. That is important because it uses interrupted time series, because there is no control unit.

So in some senses this issue has been addressed _for national level lift tests_.

## Geo-level lift tests

For geo-level lift tests, we have some geo's that are treated and some which are not. So this would now move over to being a synthetic control problem.

CausalPy now has [docs](https://causalpy.readthedocs.io/en/latest/notebooks/multi_cell_geolift.html) on multi-cell geolift test evaluation, using synthetic control. That notebook would need to be checked over, but would probably form a good basis for a how-to on the causalpy side of things. We would certainly need to add something on how to extract the results in a form that pymc-marketing wants.

Pymc-marketing also now has a good multi-dimensional / geo-level docs page [here](https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_multidimensional_example.html). Though that only has 2 geo's, presumably to keep things simple. So we would probably want a new notebook in pymc-marketing which has a richer geo-level test example. We would probably mirror the way how how it's done in the [national level lift test notebook section](https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_lift_test.html#about-lift-tests), namely we don't call causalpy, we just provide the lift test results and point readers to the appropriate causalpy docs on how to conduct the geo-lift test.
